samples/src/test/java/io/pravega/schemaregistry/samples/TestPravegaClientEndToEnd.java

```diff
@@ -0,0 +1,1353 @@
+
+        Serializer<User> deserializer = SerializerFactory.jsonMultiTypeDeserializer(serializerConfig, map);
+
+        EventStreamReader<User> reader = clientFactory.createReader("r1", rg, deserializer, ReaderConfig.builder().build());
+
+        EventRead<User> event = reader.readNextEvent(10000);
+
+        assertNotNull(event.getEvent());
+        assertTrue(event.getEvent() instanceof DerivedUser2);
+        event = reader.readNextEvent(10000);
+        assertNotNull(event.getEvent());
+        assertTrue(event.getEvent() instanceof DerivedUser1);
+        reader.close();
+        // endregion
+
+        // region read into writer schema
+        String rg2 = "rg2" + stream;
+        readerGroupManager.createReaderGroup(rg2,
+                ReaderGroupConfig.builder().stream(NameUtils.getScopedStreamName(scope, stream)).disableAutomaticCheckpoints().build());
+
+        Serializer<WithSchema<JsonNode>> genericDeserializer = SerializerFactory.jsonGenericDeserializer(serializerConfig);
+
+        EventStreamReader<WithSchema<JsonNode>> reader2 = clientFactory.createReader("r1", rg2, genericDeserializer, ReaderConfig.builder().build());
+
+        EventRead<WithSchema<JsonNode>> genEvent = reader2.readNextEvent(10000);
+        assertNotNull(genEvent.getEvent());
+        genEvent = reader2.readNextEvent(10000);
+        assertNotNull(genEvent.getEvent());
+        reader2.close();
+        // endregion
+
+        // region read using multiplexed and generic record combination
+        String rg3 = "rg3" + stream;
+        readerGroupManager.createReaderGroup(rg3,
+                ReaderGroupConfig.builder().stream(NameUtils.getScopedStreamName(scope, stream)).disableAutomaticCheckpoints().build());
+
+        Map<Class<? extends User>, JSONSchema<User>> map2 = new HashMap<>();
+        // add only one schema
+        map2.put(DerivedUser1.class, schema1);
+
+        Serializer<Either<User, WithSchema<JsonNode>>> eitherDeserializer =
+                SerializerFactory.jsonTypedOrGenericDeserializer(serializerConfig, map2);
+
+        EventStreamReader<Either<User, WithSchema<JsonNode>>> reader3 = clientFactory.createReader("r1", rg3, eitherDeserializer, ReaderConfig.builder().build());
+
+        EventRead<Either<User, WithSchema<JsonNode>>> e1 = reader3.readNextEvent(10000);
+        assertNotNull(e1.getEvent());
+        assertTrue(e1.getEvent().isRight());
+        e1 = reader3.readNextEvent(10000);
+        assertTrue(e1.getEvent().isLeft());
+        assertTrue(e1.getEvent().getLeft() instanceof DerivedUser1);
+        reader3.close();
+        readerGroupManager.close();
+        //endregion
+
+        client.removeGroup(groupId);
+        streamManager.sealStream(scope, stream);
+        streamManager.deleteStream(scope, stream);
+        streamManager.close();
+    }
+
+    @Test
+    @SneakyThrows
+    public void testMultiFormatSerializerAndDeserializer() {
```

Comment:
This fails since with the following error 

```
testMultiFormatSerializerAndDeserializer FAILED
    java.lang.IllegalStateException: Expected the service TableStore [FAILED] to be RUNNING, but the service has FAILED
```